File: fontend/src/components/Services/VocativeService.js
import axios from 'axios'

// Author: Phạm Văn Đạt
// function: Service xử lý thao tác xưng hô
//  created time: 12:06 15/08/2022
export class VocativeService{

    url= "https://localhost:44355/api/Vocative/";


    // lấy danh sách xưng hô
    async getAll(){
       try{
            return axios.get(this.url).then((res) => {
                return res.data.data;
             })
             .catch((e) => {
               console.log(e);
             });         

       }catch(error){
        console.log(error);
       }
    }

    // lấy danh sách xưng hô theo tên
    async getByName(search){
        try{
            return axios.get(this.url+search).then((res) => {
                return res.data;
             })
             .catch((e) => {
               console.log(e);
             });         
        }catch(error){
            console.log(error);
        }
    }
}

File: backend/WebApplication/WebDomain/Services/VocativeService/VocativeService.cs
using Common.Resources;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WebInfrastructure;

namespace WebDomain
{
    /// <summary>
    /// Author: Phạm Văn Đạt
    /// Created Time: 10:55 13/08/2022
    /// Service xử lý bảng xưng hô
    /// </summary>
    public class VocativeService : IVocativeService
    {
        private readonly IDapperRepository _dapper;
        public VocativeService(IDapperRepository dapper)
        {
            _dapper = dapper;
        }

        /// <summary>
        /// Author: Phạm Văn Đạt
        /// Created Time: 11:25 13/08/2022
        /// Lấy danh sách xưng hô
        /// </summary>
        /// <returns></returns>
        public async Task<ReponsitoryModel> GetAll()
        {
            try
            {
                string sql = @"SELECT VocativeId,VocativeName FROM Vocative";
                List<VocativeModel> result = await _dapper.GetAllAsync<VocativeModel>(sql);
                if (result.Count == 0)
                    return new ReponsitoryModel { Data = null, Message = MessageError.NotValue, StatusCode = CodeError.Code400 };
                return new ReponsitoryModel { Data = result, Message = MessageSuccess.GetSuccess, StatusCode = CodeSuccess.Status200 };
            }
            catch(Exception ex)
            {
                return new ReponsitoryModel { Data = ex.Message, Message = MessageError.ProcessError, StatusCode = CodeError.Code500 };
            }
        }


        /// <summary>
        /// Author: Phạm Văn Đạt
        /// Created Time: 11:25 13/08/2022
        /// Lấy danh sách xưng hô theo tên
        /// </summary>
        /// <returns></returns>
        public async Task<ReponsitoryModel> GetAllByName(string name)
        {
            try
            {
                string sql = @"SELECT VocativeId,VocativeName FROM Vocative WHERE VocativeName like @search;";
                var parameters = new Dictionary<string, object>()
                {
                    ["search"] = $"%{name}%"
                };
                var result = await _dapper.FindTAsync<VocativeModel>(sql, parameters);
                if (result == null)
                    return new ReponsitoryModel { Data = null, Message = MessageError.NotValue, StatusCode = CodeError.Code400 };
                return new ReponsitoryModel { Data = result, Message = MessageSuccess.GetSuccess, StatusCode = CodeSuccess.Status200 };
            }
            catch (Exception ex)
            {
                return new ReponsitoryModel { Data = ex, Message = MessageError.ProcessError, StatusCode = CodeError.Code500 };
            }
        }
    }
}
